5.4.10 build failure

👋 trying to build the latest release, but run into some build issue. The error log is as below:


<details>
<summary>error build log</summary>

```
  /tmp/vectorscan-20230907-31078-ev28ln/vectorscan-vectorscan-5.4.10/src/util/arch/arm/cpuid_inline.h:33:10: fatal error: 'sys/auxv.h' file not found
  #include <sys/auxv.h>
           ^~~~~~~~~~~~
  1 error generated.
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/6113993720/job/16601392063?pr=141717
relates to Homebrew/homebrew-core#141717

--- a/cmake/arch.cmake
+++ b/cmake/arch.cmake
@@ -24,6 +24,9 @@ int main() {
 endif ()
 
 if (ARCH_AARCH64)
+    if (APPLE)
+       set (FAT_RUNTIME OFF)
+    endif()
     set(PREV_FLAGS "${CMAKE_C_FLAGS}")
     if (BUILD_SVE2_BITPERM)
         set(CMAKE_C_FLAGS "-march=${GNUCC_ARCH} ${CMAKE_C_FLAGS}")

--- a/src/util/arch/arm/cpuid_inline.h
+++ b/src/util/arch/arm/cpuid_inline.h
@@ -30,7 +30,9 @@
 #ifndef AARCH64_CPUID_INLINE_H_
 #define AARCH64_CPUID_INLINE_H_
 
+#if defined(__linux__)
 #include <sys/auxv.h>
+#endif
 
 #include "ue2common.h"
 #include "util/arch/common/cpuid_flags.h"
@@ -40,6 +42,7 @@ int check_neon(void) {
     return 1;
 }
 
+#if defined(__linux__)
 static inline
 int check_sve(void) {
     unsigned long hwcap = getauxval(AT_HWCAP);
@@ -57,5 +60,16 @@ int check_sve2(void) {
     }
     return 0;
 }
+#else
+static inline
+int check_sve(void) {
+    return 0;
+}
+
+static inline
+int check_sve2(void) {
+    return 0;
+}
+#endif
 
 #endif // AARCH64_CPUID_INLINE_H_

